### ISSUE TITLE
Routing Logic

### ISSUE THREAD

**Issue by vangelov** [gothinkster/apprun-realworld-example-app#1]
Hello!

Can please help me understand how the routing actually works? I can't really find the logic which determines which view to show depending on the current location. 

Thank you!

**Comment by yysun** [comment]
Routing is event-driven. When AppRun detects the URL hash changes, it fires the events. E.g., for http://.../#/article/{slug}, it fires the # event with the _article_ and the _{slug}_ as the event parameters, so in the src/main.tsx (Line 26), the # event is converted into the /#_article_ event. It leaves the _{slug}_ as the event parameter.

``` JavaScript
app.on('#', async (route, ...p) => {
  // ...
  app.run(`#/${route || ''}`, ...p);
})
```
The /#_article_ event actives the ArticleComponent to render the screen.

``` JavaScript
import app, { Component } from 'apprun';
class ArticleComponent extends Component{
  state = {}
  view = (state) => <div></div>
  update = {
    '#/article': async(state, slug) => {}
  }
}
```
Basically, the components are sitting there and waiting for their events to come.

**Comment by vangelov** [comment]
Thank you for your answer! 

But what happens if you have two components subscribed to the same event? 
You will render both of them in this case.

**Comment by yysun** [comment]
If they are mounted to different elements. Both will render. But, if the same element, last component wins.
